lib/livebook/session.ex

```diff
@@ -814,6 +833,20 @@ defmodule Livebook.Session do
     {:noreply, handle_save_finished(state, result, file, default?)}
   end
 
+  def handle_info(:system_memory, state) do
+    {:noreply, state |> system_memory_usage() |> schedule_system_memory()}
+  end
+
+  def handle_info({:memory_usage, memory_usage}, state) do
+    Process.cancel_timer(state.system_memory_timer_ref)
+    runtime_memory = memory_usage
+    system_memory = Utils.fetch_system_memory()
+    memory = %{runtime: runtime_memory, system: system_memory}
```

Comment:
Ah, we should also define `after_operation` for `{:set_runtime, _pid, runtime}` and if runtime is `nil`, we can set runtime memory usage back to `nil`.